plugins/org.eclipse.elk.alg.common/src/org/eclipse/elk/alg/common/nodespacing/internal/algorithm/NodeLabelAndSizeUtilities.java

```diff
@@ -83,9 +86,150 @@ public static void setupNodePaddingForPortsWithOffset(final NodeContext nodeCont
+
+                case EAST:
+                    insidePartIsBigger = insidePart > nodeCellPadding.right;
+                    nodeCellPadding.right = Math.max(nodeCellPadding.right, insidePart);
+                    if (symmetry && insidePartIsBigger) {
+                        nodeCellPadding.right = Math.max(nodeCellPadding.left, nodeCellPadding.right);
+                        // For symmetry, the portBorderOffset is not considered (only label + label padding)
+                        nodeCellPadding.left = nodeCellPadding.right + portBorderOffset;
+
+                    }
+                    break;
+
+                case WEST:
+                    insidePartIsBigger = insidePart > nodeCellPadding.left;
+                    nodeCellPadding.left = Math.max(nodeCellPadding.left, insidePart);
+                    if (symmetry && insidePartIsBigger) {
+                        nodeCellPadding.left = Math.max(nodeCellPadding.left, nodeCellPadding.right);
+                        // For symmetry, the portBorderOffset is not considered (only label + label padding)
+                        nodeCellPadding.right = nodeCellPadding.left + portBorderOffset;
+                    }
+                    break;
+                }
+            }
         }
     }
     
+    /**
+     * Compute the part of the labels of the port that is inside the node.
+     * 
+     * @param port
+     *            The port
+     * @param portBorderOffset
+     *            The border offset of the port
+     * @param gap
+     *            gap between labels.
+     * @return the part of the labels that is inside the node
+     */
+    private static double computeInsidePart(final PortAdapter<?> port, final double portBorderOffset,
+            final double gap) {
+        double insidePart = 0;
+        for (LabelAdapter<?> label : port.getLabels()) {
+            // TODO : handle case of port with several labels according to the location of labels. Only one label case
+            // has been tested.
+            // if (insidePart > 0) {
+            // // This is not the first label, add margin
+            // insidePart += gap;
+            // }
+            // insidePart += computeInsidePart(label, port, portBorderOffset);
+            insidePart = computeInsidePart(label, port, portBorderOffset);
+        }
+        return insidePart;
+    }
+
+
+    /**
+     * Compute the part of the label of the port that is inside the node.
+     * 
+     * @param label
+     *            The label
+     * @param port
+     *            The port
+     * @param portBorderOffset
+     *            The border offset of the port
+     * @return the part of the label that is inside the node
+     */
+    public static double computeInsidePart(final LabelAdapter<?> label, final PortAdapter<?> port,
+            final double portBorderOffset) {
+        return computeInsidePart(label.getPosition().x, label.getPosition().y, label.getSize().x, label.getSize().y,
```

Comment:
Maybe pass the `KVector`s here to reduce the number of parameters.

Reply:
Yes. computeInsidePart is also used in org.eclipse.elk.alg.layered.graph.transform.ElkGraphImporter where we have not access to a KVector (but we can create them).